src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import { DataViewerProposal } from './interfaces/DataViewerProposal';
+
+/**
+   * Safely gets the proposal value for a given proposal attribute key.
+   *
+   * @param  {DataViewerProposal} proposal
+   * @param  {string} key The key for the proposal attribute we want.
+   * @return {(Array|undefined)}
+   */
+const getProposalValue = (proposal: DataViewerProposal, key: string) => {
+  // This is a temporary fix:
+  // The API currently allows fields to be set to blank strings,
+  // and dutifully includes those in the value array.
+  // We do not consider those valid when choosing a proposal value,
+  // so here we are making sure we have at least one non-blank value.
+  if (!proposal.values[key]?.some((value) => value !== '')) {
+    return undefined;
+  }
+
+  return proposal.values[key];
+};
+
+/**
+   * From a list of proposal attribute keys, safely gets the first valid proposal value.
+   *
+   * Key order is relevant, as the first matching key's value will be returned.
+   *
+   * @param  {DataViewerProposal} proposal
+   * @param  {string[]} keys An array of keys, ordered by preference with most-preferred first.
+   * @return {(Array|undefined)} The proposal value array. Falls back to undefined.
+   */
+const getProposalValueFromCandidates = (proposal: DataViewerProposal, keys: string[]) => {
```

Comment:
Since this returns an array (for the case where there multiple answers to a single form field), should this be `getProposalValuesFromCandidates` (emphasis on the plural) or so?

Reply:
Yes, it should! And I guess if `getProposalValue` hangs around (tbd) it should be renamed, too.